community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -377,6 +382,85 @@ public Object getProperty( String key, Object defaultValue )
     }
 
     @Override
+    public Map<String, Object> getProperties( String... keys )
+    {
+        try ( Statement statement = actions.statement() )
+        {
+            try ( Cursor<NodeItem> node = statement.readOperations().nodeCursor( nodeId ) )
+            {
+                if ( !node.next() )
+                {
+                    throw new NotFoundException( "Node not found",
+                            new EntityNotFoundException( EntityType.NODE, getId() ) );
+                }
+
+                try ( Cursor<PropertyItem> propertyCursor = node.get().properties() )
+                {
+                    Map<String, Object> properties = new HashMap<>( ((int) (keys.length * 1.3f)) );
+
+                    // Specific properties given
+                    int[] propertyKeys = new int[keys.length];
+                    for ( int i = 0; i < keys.length; i++ )
+                    {
+                        String key = keys[i];
+                        propertyKeys[i] = statement.readOperations().propertyKeyGetForName( key );
+                    }
```

Comment:
So you think it would be better to do the name->int lookup on every check instead? Won't that impact performance a bit too much?